### PR TITLE
Ui profile subjects tab

### PR DESCRIPTION
**Pull request by CheskaG**
image

### CODE REVIEW

**Comment by EdasL** [comment]
Would like to see some indication if the subject is learnt or not

**Comment by CheskaG** [comment]
subjects
edit

File: src/pages/Profile/Profile.tsx [hunk 1]
```diff
@@ -63,21 +77,29 @@ const Profile: FC = () => {
     )
   }
 
+  function renderEditButton() {
+    if (isCurrentUser)
+      return <Chip className={classes.editButton} color="primary" label="Edit Info" />
+  }
```
Comment (EdasL) [inline src/pages/Profile/Profile.tsx:83]: Screenshot 2020-05-23 at 13 10 03
```suggestion
  function renderEditButton() {
    if (!isCurrentUser) return null
    
    return <Chip className={classes.editButton} color="primary" label="Edit Info" />
  }
```

File: src/pages/Profile/Profile.tsx [hunk 2]
```diff
@@ -10,24 +10,38 @@ import {
...
-    firstName: 'jeff',
-    lastName: 'jefferson',
-    email: 'yo@gmail.com',
-    team: { id: 1, name: 'genericTeam' },
-    teamId: 1,
-    role: UserRole.Junior,
-  }
+  const [value, setValue] = useState(0)
+  const [userSubjects, setUserSubjects] = useState<Array<UserSubjectType> | null>(null)
+  const [isCurrentUser, setIsCurrentUser] = useState<boolean>(true)
```
Comment (EdasL) [inline src/pages/Profile/Profile.tsx:29]: Screenshot 2020-05-23 at 13 09 11

File: src/App.tsx [hunk 3]
```diff
@@ -101,6 +101,12 @@ const App: FC = () => {
             <PrivateRoute isLoggedOn={isLoggedOn} path={routes.HOME}>
               <Calendar />
             </PrivateRoute>
+            <PrivateRoute isLoggedOn={isLoggedOn} path={routes.USER}>
+              <Profile />
+            </PrivateRoute>
+            <PrivateRoute isLoggedOn={isLoggedOn} path={routes.TEAM}>
+              <Team />
+            </PrivateRoute>
```
Comment (EdasL) [inline src/App.tsx:109]:
```suggestion
            <PrivateRoute isLoggedOn={isLoggedOn} path={routes.USER}>
              <Profile />
            </PrivateRoute>
            <PrivateRoute isLoggedOn={isLoggedOn} path={routes.TEAM}>
              <Team />
            </PrivateRoute>
                        <PrivateRoute isLoggedOn={isLoggedOn} path={routes.HOME}>
              <Calendar />
            </PrivateRoute>
```

File: src/data/api/subjects.ts [hunk 4]
```diff
@@ -1,6 +1,5 @@
 import { Subject, UserSubject } from 'types/models/subject'
 
-import { UserSubject } from 'types/models/userSubject'
```
Comment (EdasL) [inline src/data/api/subjects.ts]: Don't remove this, it is used

File: src/pages/Profile/Profile.tsx [hunk 5]
```diff
@@ -63,7 +100,23 @@ const Profile: FC = () => {
...
+    return <Chip className={classes.editButton} color="primary" label="Edit Info" />
+  }
+
+  function getUserRoleString(role: number) {
+    if (role === UserRole.Mid) return 'Mid'
+    if (role === UserRole.Senior) return 'Senior'
+    if (role === UserRole.TeamLead) return 'TeamLead'
+    if (role === UserRole.God) return 'God'
+    return 'Junior'
+  }
```
Comment (EdasL) [inline src/pages/Profile/Profile.tsx:115]: How about using switch? 
```suggestion
  function getUserRoleString(role: number) {
  switch (role) {
    case UserRole.Mid:
        return 'Mid'
     case UserRole.Senior:
        return 'Senior'
     case UserRole.TeamLead:
        return 'TeamLead'
     case UserRole.God:
        return 'God'
      default:
        return 'Junior'
  }
  }
```

File: src/pages/Profile/Profile.tsx [hunk 6]
```diff
@@ -97,10 +151,35 @@ const Profile: FC = () => {
...
+      return (
+        <GridList cellHeight={45} cols={1} className={classes.gridList}>
+          {userSubjects.map(({ id: userSubjectId, subject, isLearned }) => (
+            <GridListTile key={userSubjectId}>
+              {renderChip(subject, isLearned, userSubjectId)}
+            </GridListTile>
+          ))}
+        </GridList>
+      )
+    return null
```
Comment (EdasL) [inline src/pages/Profile/Profile.tsx:177]:
```suggestion
    if (!userSubjects) return null
    
    return (
        <GridList cellHeight={45} cols={1} className={classes.gridList}>
          {userSubjects.map(({ id: userSubjectId, subject, isLearned }) => (
            <GridListTile key={userSubjectId}>
              {renderChip(subject, isLearned, userSubjectId)}
            </GridListTile>
          ))}
        </GridList>
      )
```

**Pull request merged**